src/Type/Constant/ConstantStringType.php

```diff
@@ -140,7 +141,7 @@ public function isSuperTypeOf(Type $type): TrinaryLogic
 			return $this->value === $type->value ? TrinaryLogic::createYes() : TrinaryLogic::createNo();
 		}
 
-		if ($type instanceof parent) {
+		if ($type instanceof parent || $type instanceof CallableType) {
```

Comment:
oh wow.. turns out the fix itself is pretty simple and required at a different spot I initially tried ;-).

good job!

Reply:
the work here is not done yet, and I'm still not sure if that's the right fix :)